localsubscriber.go

```diff
@@ -80,25 +70,16 @@ func (s *LocalSubscriber) Dispatch(u *Update, fromHistory bool) bool {
 
 // Ready flips the ready flag to true and flushes queued live updates returning number of events flushed.
 func (s *LocalSubscriber) Ready() (n int) {
-	s.liveMutex.Lock()
-	s.outMutex.Lock()
-
 	for _, u := range s.liveQueue {
 		select {
 		case s.out <- u:
 			n++
 		default:
-			s.handleFullChan()
-			s.liveMutex.Unlock()
-
 			return n
```

Comment:
`handleFullChan` was calling `atomic.StoreInt32(&s.disconnected, 1)`. Removing this call changes the behavior. 

Could this not cause events to be silently unflushed when the channel is full, causing the client to enter a ready state after missing events?

Reply:
In the meantime, I changed the code to use the context API instead, simplifying the code and fixing the race conditions showcased by the benchmark.

The existing behavior should be preserved, too.